model/events/LtiAgsListener.php

```diff
@@ -81,19 +83,32 @@ public function onDeliveryExecutionStateUpdate(DeliveryExecutionState $event)
         }
     }
 
+    public function onDeliveryExecutionResultsRecalculated(DeliveryExecutionResultsRecalculated $event)
+    {
+        $deliveryExecution = $event->getDeliveryExecution();
+
+        if ($launchData = $this->getLtiContextRepository()->findByDeliveryExecution($deliveryExecution)) {
```

Comment:
Maybe add some check for $launchData e.g. instanceof or something. You just put it straight to method and it can easily burn and crash here.

Reply:
I checked and this method could return null and then ...

Reply:
If `findByDeliveryExecution` will return anything except Launch data or null it will crash by type error, no need for double check

Reply:
If thats ok for you, I would rather handle my errors somehow.